### PR TITLE
set value to default if zero or less for idle_timeout and entries_per_pa...

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -91,6 +91,16 @@ def save_as_settings_type
     end
   end
 
+  def ensure_values_not_zero
+    # set value to default if zero or negative
+    # id => 4, name => entries_per_page - if zero, error FloatDomainError Infinity
+    # id => 7, name => idle_timeout - if zero,user is logged out immediately
+    # id => 18, name => puppet_interval
+    if [4,7,18].include?(self.id) && self.value <= 0
```
Comment (ohadlevy) [inline app/models/setting.rb]: Are you sure ids are always the same? you might need to go by name here

Comment (isratrade) [inline app/models/setting.rb]: You're right. I'll make the changes tomorrow.

Comment (ohadlevy) [inline app/models/setting.rb]: thanks, I also think tests are required for this specific case.

**Comment by isratrade** [comment]
I used validations rather than a before_save callback method to ensure  idle_timeout and entries_per_page are not zero.
The tests pass sometimes and other times, I get:
TypeError: instance of IO needed

File: app/models/setting.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class Setting < ActiveRecord::Base
   validates_presence_of :default, :unless => Proc.new { |s| !s.default } # broken validator
   validates_uniqueness_of :name
   validates_numericality_of :value, :if => Proc.new {|s| s.settings_type == "integer"}
-  validates_numericality_of :value, :if => Proc.new {|s| s.name == "puppet_interval"}, :greater_than => 0
+  validates_numericality_of :value, :if => Proc.new {|s| %w(puppet_interval idle_timeout entries_per_page).include?(s.name) }, :greater_than => 0
```
Comment (ohadlevy) [inline app/models/setting.rb]: looks good, for better readability, I would consider moving the setting names into a constant or class method?

File: test/unit/setting_test.rb [hunk 3]
```diff
@@ -70,4 +70,22 @@ def test_boolean_values_should_have_setting_type_for_true
     assert_equal true, Setting["oo"]
   end
 
+  test "idle_timeout should not be zero" do
+    setting = Setting.new(:name => 'idle_timeout', :value => 0, :default => 60 )
```
Comment (ohadlevy) [inline test/unit/setting_test.rb]: afaik, the validation would fail here regardless as such a setting already exists

**Comment by ohadlevy** [comment]
TypeError: instance of IO needed usually happens with YAML.load gets an stream, not sure why it happens in this case

**Comment by isratrade** [comment]
TypeError: instance of IO error is now gone.

**Comment by ohadlevy** [comment]
actually, puppet_interval cant be lower then 10, see http://theforeman.org/issues/1475.

i'll merge anyway, but a fix fr 1475 is needed regardless

**Pull request closed without merging**